src/axom/primal/geometry/Polygon.hpp

```diff
@@ -235,17 +365,20 @@ class Polygon
    * Initial check is that the polygon has three or more vertices
    * \return True, if the polygon is valid, False otherwise
    */
-  bool isValid() const { return m_vertices.size() >= 3; }
+  AXOM_HOST_DEVICE
+  bool isValid() const { return m_num_vertices >= 3; }
 
 private:
-  axom::Array<PointType> m_vertices;
+  ArrayType m_vertices;
+  int m_num_vertices {0};
```

Comment:
This PR is looking good to me. I think we could eliminate m_num_vertices in the dynamic case with a small change. Suppose the static case introduced a subclass of StackArray that included the length and a few more Array-like niceties. The m_num_vertices member gets moved into that class and the addition of a few methods like size/clear/push_back lets some of the std::enable_ifs in Polygon go away too if the static case for "using ArrayType" uses the class below.

```
template <typename T, int N>
struct StaticArray : public StackArray<T, N>
{
   // constructors...

   int size() const { return m_size; }
   void push_back(const T &obj) { m_data[m_size++] = obj; }
   void clear() { m_size = 0; }
private:
   int m_size {0};
};
```

Reply:
👍 Added a `StaticArray`, removing the need for `m_num_vertices`. Was able to clean up some of the std::enable_ifs, but still need to keep some around to keep the host device warnings away.